modules/tinymce/src/plugins/table/test/ts/browser/ui/TableClassListButtonsTest.ts

```diff
@@ -0,0 +1,121 @@
+import { Assertions, Keys, Waiter } from '@ephox/agar';
+import { afterEach, describe, it } from '@ephox/bedrock-client';
+import { Arr } from '@ephox/katamari';
+import { TinyHooks, TinySelections, TinyUiActions } from '@ephox/mcagar';
+import { SugarBody, SugarElement } from '@ephox/sugar';
+import { assert } from 'chai';
+import Editor from 'tinymce/core/api/Editor';
+import { EditorEvent } from 'tinymce/core/api/util/EventDispatcher';
+import { TableModifiedEvent } from 'tinymce/plugins/table/api/Events';
+import Plugin from 'tinymce/plugins/table/Plugin';
+import Theme from 'tinymce/themes/silver/Theme';
+
+describe('browser.tinymce.plugins.table.ui.TableClassListButtonsTest', () => {
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    plugins: 'table',
+    indent: false,
+    table_class_list: [
+      {
+        title: 'A',
+        value: 'a'
+      },
+      {
+        title: 'B',
+        value: 'b'
+      }
+    ],
+    toolbar: 'tableclass',
+    base_url: '/project/tinymce/js/tinymce',
+    setup: (editor: Editor) => {
+      editor.on('tablemodified', logEvent);
+    }
+  }, [ Plugin, Theme ], true);
+
+  let events: Array<EditorEvent<TableModifiedEvent>> = [];
+  const logEvent = (event: EditorEvent<TableModifiedEvent>) => {
+    events.push(event);
+  };
+
+  afterEach(() => {
+    events = [];
+  });
+
+  const setEditorContentTableAndSelection = (editor: Editor) => {
+    editor.setContent(
+      '<table>' +
+        '<tbody>' +
+          '<tr>' +
+            '<td>Filler</td>' +
+          '</tr>' +
+        '</tbody>' +
+      '</table>'
+    );
+
+    TinySelections.setSelection(editor, [ 0, 0, 0 ], 0, [ 0, 0, 0 ], 1);
+  };
+
+  const openMenu = (editor: Editor) =>
+    TinyUiActions.clickOnToolbar(editor, 'button[title="Table styles"]');
+
+  const closeMenu = (editor) =>
+    TinyUiActions.keydown(editor, Keys.escape());
+
+  const checkMenuPresence = async (editor: Editor, label: string, expected: Record<string, number>, container: SugarElement<HTMLElement>) => {
```

Comment:
```suggestion
  const pCheckMenuPresence = async (editor: Editor, label: string, expected: Record<string, number>, container: SugarElement<HTMLElement>) => {
```
Helpers in tests currently use the p prefix when it's async, so we should keep that convention for now. There also other helper functions here, so please make sure to update all of them.